h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/rabit/RabitTrackerH2O.java

```diff
@@ -71,26 +71,28 @@ public boolean start(long timeout) {
             throw new IllegalStateException("Rabit tracker already started.");
         }
         RabitTrackerH2OThread trackerThread = new RabitTrackerH2OThread(this);
-        trackerThread.setDaemon(true);
-        trackerThread.start();
         this.trackerThread = trackerThread;
+        trackerThread.start();
         return true;
     }
 
     @Override
     public void stop() {
-        if(null != this.trackerThread) {
-            this.trackerThread.interrupt();
+        assert this.trackerThread != null;
```

Comment:
Null check removed. This is never assigned null before this method's invocation now.

In fact, it should never ever be null at this point. That's why there is this new assert.